bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -918,17 +927,22 @@ void updatePrimaryStar() {
             getSelectedGroup().foreach(igg -> {
                 final TargetEnvironment env = _dataObject.getTargetEnvironment();
                 final GuideGroup primary = env.getOrCreatePrimaryGuideGroup();
-                if (primary != igg.group() && confirmGroupChange(primary, igg.group())) {
-                    final GuideEnvironment ge = env.getGuideEnvironment();
-                    if (ge != null) {
-                        _dataObject.setTargetEnvironment(env.setGuideEnvironment(ge.setPrimaryIndex(igg.index())));
-
-                        // If we are switching to an automatic group, we also
-                        // possibly need to update the position angle.
-                        final GuideGrp grp = igg.group().grp();
-                        if (grp instanceof AutomaticGroup.Active) {
-                            updatePosAngle(((AutomaticGroup.Active) grp).posAngle());
-                        }
+
+                // If the auto group is disabled and set to primary, then make it an initial auto group.
+                final GuideGrp grp = igg.group().grp();
+                final GuideEnvironment ge = env.getGuideEnvironment();
```

Comment:
Can we wrap `ge` in an `Option/Optional` and use a foreach to avoid the null check?